chore: removing legacy REST client generation

bensooraj opened the pull request:
Hey!

This is an attempt at resolving tendermint/starport#1066.

#### What I have done:
- Deleted the plush templates which contribute to `x/foo/client/rest`:
   - starport/templates/typed/stargate/component/x/{{moduleName}}/**client/rest/query{{TypeName}}.go**.plush
   - starport/templates/typed/stargate/messages/x/{{moduleName}}/**client/rest/tx{{TypeName}}.go**.plush
   - starport/templates/module/create/stargate/x/{{moduleName}}/**client/rest/rest.go**.plush

- Modified `module.go`'s template file `starport/templates/module/create/stargate/x/{{moduleName}}/module.go.plush`, such that `AppModuleBasic`s method `RegisterRESTRoutes` does nothing. I didn't delete this method because `AppModuleBasic` interface defined in the cosmos-idk requires it:

https://github.com/cosmos/cosmos-sdk/blob/711976efc13f135d334e41cd4a61bb490a8befa3/types/module/module.go#L46-L60

- Verified that the
   - scaffolded `app` and `type` build without errors using `starport build`
   - transactions and queries are working as expected from the generated CLI client
 
#### What I am struggling with:

When I start the application using `starport serve`, the project builds successfully without any errors but the frontend application doesn't start. I might need some help with this.

Thanks!

lubtd commented:
Hey @bensooraj thanks for the PR 👍 REST routes are actually deprecated and we must remove them.

If we want to do it completely, we can even remove `x/{{moduleName}}/keeper/query.go.plush` when scaffolding a module and `component/x/{{moduleName}}/keeper/query_{{typeName}}.go.plush` when scaffolding a type. Those are actually the legacy way to handle queries with REST routes. Queries with grpc are handled in `grpc_query_...go`

bensooraj commented:
@lubtd aah okay! Thanks for the feedback. On it!

bensooraj commented:
Hey @lubtd,

I have removed the following two templates:
- x/{{moduleName}}/keeper/query.go.plush
- component/x/{{moduleName}}/keeper/query_{{typeName}}.go.plush

I have modified the `LegacyQuerierHandler` in `starport/templates/module/create/stargate/x/{{moduleName}}/module.go.plush` such that it returns `nil`. Not 100% sure if this is the standard approach. Please let me know if the approach is incorrect.

And finally, the UI is still broken. I tried running `npm i` inside the `vue` directory, but it threw the following error:
```sh
# npm i
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: @starport/template@0.1.44
npm ERR! Found: @cosmjs/encoding@0.24.1
npm ERR! node_modules/@cosmjs/encoding
npm ERR!   @cosmjs/encoding@"0.24.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @cosmjs/encoding@"0.25.0-alpha.2" from @starport/vue@0.1.47
npm ERR! node_modules/@starport/vue
npm ERR!   @starport/vue@"^0.1.44" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /Users/<username>/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/<username>/.npm/_logs/2021-05-04T10_48_42_891Z-debug.log
```

I have a feeling that this has to do with #1076.

fadeev commented:
@bensooraj yes, we probably should ignore the frontend for now. It's broken on `develop` due to dependency issues. @clockworkgr will investigate it soon. Removing REST endpoints shouldn't affect the frontend anyway.

lubtd reviewed (approved):
I can scaffold an app and create types without side effects 👍

Thanks for the change

fadeev reviewed (approved):
Scaffolding types and modules works for me. Thanks so much for contributing to Starport, @bensooraj! 👏

bensooraj commented:
> Scaffolding types and modules works for me. Thanks so much for contributing to Starport, @bensooraj! 👏

Thank you so much @fadeev and @lubtd!

Hope I can contribute to the cosmos ecosystem more 😄

ilgooz commented:
Perfect, thanks @bensooraj!
